src/guarneri/_typing.py

```diff
@@ -0,0 +1,18 @@
+from abc import abstractmethod
+from typing import Protocol, Sequence, Union, runtime_checkable
+
+
+@runtime_checkable
+class Device(Protocol):
```

Comment:
Seems redundant with `helpers.Device`. Maybe we should pick one and decide what *exactly* a "device" is in this context?

Reply:
helpers.py defines a `Device` type alias as either an ophyd or ophyd async Device.

```python
Device: TypeAlias = AsyncDevice | ThreadedDevice
```

_typing.py defines a `Device` protocol as anything that has the name property.

I think we should settle on a clear concept of what guarneri considers to be a "device" in the general sense.

I don't think we need to do this now, though. I opened an issue and we can decide later: https://github.com/BCDA-APS/guarneri/issues/28